spring-rabbit-stream/src/main/java/org/springframework/rabbit/stream/micrometer/RabbitStreamMessageReceiverContext.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.rabbit.stream.micrometer;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+import org.springframework.amqp.rabbit.support.micrometer.RabbitListenerObservation;
+import org.springframework.amqp.rabbit.support.micrometer.RabbitListenerObservationConvention;
+import org.springframework.amqp.rabbit.support.micrometer.RabbitMessageReceiverContext;
+
+import com.rabbitmq.stream.Message;
+import io.micrometer.common.KeyValues;
+import io.micrometer.observation.transport.ReceiverContext;
+
+/**
+ * {@link ReceiverContext} for stream {@link Message}s.
+ *
+ * @author Gary Russell
+ * @since 3.0.5
+ *
+ */
+public class RabbitStreamMessageReceiverContext extends ReceiverContext<Message> {
+
+	private final String listenerId;
+
+	private final Message message;
+
+	private final String stream;
+
+	public RabbitStreamMessageReceiverContext(Message message, String listenerId, String stream) {
+		super((carrier, key) -> {
+			Map<String, Object> props = carrier.getApplicationProperties();
+			if (props != null) {
+				Object value = carrier.getApplicationProperties().get(key);
+				if (value instanceof String string) {
+					return string;
+				}
+				else if (value instanceof byte[] bytes) {
+					return new String(bytes, StandardCharsets.UTF_8);
+				}
+			}
+			return null;
+		});
+		setCarrier(message);
+		this.message = message;
+		this.listenerId = listenerId;
+		this.stream = stream;
+		setRemoteServiceName("RabbitMQ");
```

Comment:
Don't we need to have it as more granual:  `RabbitMQ Streams`?
And the same for those ` "spring.rabbit.listener" ->  "spring.rabbit..stream.listener"` ?